Synthesis failed for AccessApproval

Hello! Autosynth couldn't regenerate AccessApproval. :broken_heart:

Here's the output from running `synth.py`:

```
othing to commit, working tree clean
2020-08-12 16:45:11,270 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp7g7a924g/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp7g7a924g/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 28 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

23:45:40.201 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1

23:45:40.534 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1

23:45:40.737 [info]  FOUND: https://accessapproval.googleapis.com/$discovery/rest?version=v1
Revision check: old=20200726, new=20200807, generating=true
Creating leading directories
Writing AccessApprovalSettings to clients/access_approval/lib/google_api/access_approval/v1/model/access_approval_settings.ex.
Writing AccessLocations to clients/access_approval/lib/google_api/access_approval/v1/model/access_locations.ex.
Writing AccessReason to clients/access_approval/lib/google_api/access_approval/v1/model/access_reason.ex.
Writing ApprovalRequest to clients/access_approval/lib/google_api/access_approval/v1/model/approval_request.ex.
Writing ApproveApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/approve_approval_request_message.ex.
Writing ApproveDecision to clients/access_approval/lib/google_api/access_approval/v1/model/approve_decision.ex.
Writing DismissApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_approval_request_message.ex.
Writing DismissDecision to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_decision.ex.
Writing Empty to clients/access_approval/lib/google_api/access_approval/v1/model/empty.ex.
Writing EnrolledService to clients/access_approval/lib/google_api/access_approval/v1/model/enrolled_service.ex.
Writing ListApprovalRequestsResponse to clients/access_approval/lib/google_api/access_approval/v1/model/list_approval_requests_response.ex.
Writing ResourceProperties to clients/access_approval/lib/google_api/access_approval/v1/model/resource_properties.ex.
Writing Folders to clients/access_approval/lib/google_api/access_approval/v1/api/folders.ex.
Writing Organizations to clients/access_approval/lib/google_api/access_approval/v1/api/organizations.ex.
Writing Projects to clients/access_approval/lib/google_api/access_approval/v1/api/projects.ex.
Writing connection.ex.
Writing metadata.ex.
Writing mix.exs
Writing README.md
Writing LICENSE
Writing .gitignore
Writing config/config.exs
Writing test/test_helper.exs

23:45:40.818 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1beta1

23:45:41.025 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1beta1
{:error, "All formats failed"}
** (File.Error) could not read file "/workspace/specifications/gdd/AccessApproval-v1beta1.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-08-12 16:45:41,625 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp7g7a924g/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp7g7a924g/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp7g7a924g/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AccessApproval']' returned non-zero exit status 1.
2020-08-12 16:45:41,974 autosynth [ERROR] > Synthesis failed
2020-08-12 16:45:41,974 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 690, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 539, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 630, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/access_approval/synth.metadata', 'synth.py', '--', 'AccessApproval']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/3a29e810-6da2-4250-8b8b-12bf0c0b4f02/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AccessApproval. :sob:

Here's the output from running `synth.py`:

```
home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-accessapproval
nothing to commit, working tree clean
2020-08-12 22:29:12,498 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpvm_huhij/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpvm_huhij/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 28 files (.ex)
Generated google_apis app

05:29:44.610 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1

05:29:44.947 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1

05:29:45.154 [info]  FOUND: https://accessapproval.googleapis.com/$discovery/rest?version=v1
Revision check: old=20200726, new=20200807, generating=true
Creating leading directories
Writing AccessApprovalSettings to clients/access_approval/lib/google_api/access_approval/v1/model/access_approval_settings.ex.
Writing AccessLocations to clients/access_approval/lib/google_api/access_approval/v1/model/access_locations.ex.
Writing AccessReason to clients/access_approval/lib/google_api/access_approval/v1/model/access_reason.ex.
Writing ApprovalRequest to clients/access_approval/lib/google_api/access_approval/v1/model/approval_request.ex.
Writing ApproveApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/approve_approval_request_message.ex.
Writing ApproveDecision to clients/access_approval/lib/google_api/access_approval/v1/model/approve_decision.ex.
Writing DismissApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_approval_request_message.ex.
Writing DismissDecision to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_decision.ex.
Writing Empty to clients/access_approval/lib/google_api/access_approval/v1/model/empty.ex.
Writing EnrolledService to clients/access_approval/lib/google_api/access_approval/v1/model/enrolled_service.ex.
Writing ListApprovalRequestsResponse to clients/access_approval/lib/google_api/access_approval/v1/model/list_approval_requests_response.ex.
Writing ResourceProperties to clients/access_approval/lib/google_api/access_approval/v1/model/resource_properties.ex.
Writing Folders to clients/access_approval/lib/google_api/access_approval/v1/api/folders.ex.
Writing Organizations to clients/access_approval/lib/google_api/access_approval/v1/api/organizations.ex.
Writing Projects to clients/access_approval/lib/google_api/access_approval/v1/api/projects.ex.
Writing connection.ex.
Writing metadata.ex.
Writing mix.exs
Writing README.md
Writing LICENSE
Writing .gitignore
Writing config/config.exs
Writing test/test_helper.exs

05:29:45.235 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1beta1

05:29:45.437 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1beta1
{:error, "All formats failed"}
** (File.Error) could not read file "/workspace/specifications/gdd/AccessApproval-v1beta1.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-08-12 22:29:46,099 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpvm_huhij/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpvm_huhij/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpvm_huhij/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AccessApproval']' returned non-zero exit status 1.
2020-08-12 22:29:46,472 autosynth [ERROR] > Synthesis failed
2020-08-12 22:29:46,472 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 690, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 539, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 630, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/access_approval/synth.metadata', 'synth.py', '--', 'AccessApproval']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/47ae6257-bab6-4f70-8bc4-147aa77ff731/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AccessApproval. :sob:

Here's the output from running `synth.py`:

```
home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-accessapproval
nothing to commit, working tree clean
2020-08-13 05:03:11,499 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp1dohj8h4/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp1dohj8h4/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 28 files (.ex)
Generated google_apis app

12:03:41.972 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1

12:03:42.291 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1

12:03:42.513 [info]  FOUND: https://accessapproval.googleapis.com/$discovery/rest?version=v1
Revision check: old=20200726, new=20200807, generating=true
Creating leading directories
Writing AccessApprovalSettings to clients/access_approval/lib/google_api/access_approval/v1/model/access_approval_settings.ex.
Writing AccessLocations to clients/access_approval/lib/google_api/access_approval/v1/model/access_locations.ex.
Writing AccessReason to clients/access_approval/lib/google_api/access_approval/v1/model/access_reason.ex.
Writing ApprovalRequest to clients/access_approval/lib/google_api/access_approval/v1/model/approval_request.ex.
Writing ApproveApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/approve_approval_request_message.ex.
Writing ApproveDecision to clients/access_approval/lib/google_api/access_approval/v1/model/approve_decision.ex.
Writing DismissApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_approval_request_message.ex.
Writing DismissDecision to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_decision.ex.
Writing Empty to clients/access_approval/lib/google_api/access_approval/v1/model/empty.ex.
Writing EnrolledService to clients/access_approval/lib/google_api/access_approval/v1/model/enrolled_service.ex.
Writing ListApprovalRequestsResponse to clients/access_approval/lib/google_api/access_approval/v1/model/list_approval_requests_response.ex.
Writing ResourceProperties to clients/access_approval/lib/google_api/access_approval/v1/model/resource_properties.ex.
Writing Folders to clients/access_approval/lib/google_api/access_approval/v1/api/folders.ex.
Writing Organizations to clients/access_approval/lib/google_api/access_approval/v1/api/organizations.ex.
Writing Projects to clients/access_approval/lib/google_api/access_approval/v1/api/projects.ex.
Writing connection.ex.
Writing metadata.ex.
Writing mix.exs
Writing README.md
Writing LICENSE
Writing .gitignore
Writing config/config.exs
Writing test/test_helper.exs

12:03:42.596 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1beta1

12:03:42.814 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1beta1
{:error, "All formats failed"}
** (File.Error) could not read file "/workspace/specifications/gdd/AccessApproval-v1beta1.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-08-13 05:03:43,457 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp1dohj8h4/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp1dohj8h4/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp1dohj8h4/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AccessApproval']' returned non-zero exit status 1.
2020-08-13 05:03:43,898 autosynth [ERROR] > Synthesis failed
2020-08-13 05:03:43,899 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 690, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 539, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 630, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/access_approval/synth.metadata', 'synth.py', '--', 'AccessApproval']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/33a8a026-5614-4f09-9464-20a8cfa08b41/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AccessApproval. :sob:

Here's the output from running `synth.py`:

```
home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-accessapproval
nothing to commit, working tree clean
2020-08-14 05:03:10,758 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpd1zsb64i/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpd1zsb64i/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 28 files (.ex)
Generated google_apis app

12:03:40.597 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1

12:03:40.905 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1

12:03:41.112 [info]  FOUND: https://accessapproval.googleapis.com/$discovery/rest?version=v1
Revision check: old=20200726, new=20200807, generating=true
Creating leading directories
Writing AccessApprovalSettings to clients/access_approval/lib/google_api/access_approval/v1/model/access_approval_settings.ex.
Writing AccessLocations to clients/access_approval/lib/google_api/access_approval/v1/model/access_locations.ex.
Writing AccessReason to clients/access_approval/lib/google_api/access_approval/v1/model/access_reason.ex.
Writing ApprovalRequest to clients/access_approval/lib/google_api/access_approval/v1/model/approval_request.ex.
Writing ApproveApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/approve_approval_request_message.ex.
Writing ApproveDecision to clients/access_approval/lib/google_api/access_approval/v1/model/approve_decision.ex.
Writing DismissApprovalRequestMessage to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_approval_request_message.ex.
Writing DismissDecision to clients/access_approval/lib/google_api/access_approval/v1/model/dismiss_decision.ex.
Writing Empty to clients/access_approval/lib/google_api/access_approval/v1/model/empty.ex.
Writing EnrolledService to clients/access_approval/lib/google_api/access_approval/v1/model/enrolled_service.ex.
Writing ListApprovalRequestsResponse to clients/access_approval/lib/google_api/access_approval/v1/model/list_approval_requests_response.ex.
Writing ResourceProperties to clients/access_approval/lib/google_api/access_approval/v1/model/resource_properties.ex.
Writing Folders to clients/access_approval/lib/google_api/access_approval/v1/api/folders.ex.
Writing Organizations to clients/access_approval/lib/google_api/access_approval/v1/api/organizations.ex.
Writing Projects to clients/access_approval/lib/google_api/access_approval/v1/api/projects.ex.
Writing connection.ex.
Writing metadata.ex.
Writing mix.exs
Writing README.md
Writing LICENSE
Writing .gitignore
Writing config/config.exs
Writing test/test_helper.exs

12:03:41.193 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/GOOGLE_REST_SIMPLE_URI?version=v1beta1

12:03:41.397 [info]  FETCHING: https://accessapproval.googleapis.com/$discovery/rest?version=v1beta1
{:error, "All formats failed"}
** (File.Error) could not read file "/workspace/specifications/gdd/AccessApproval-v1beta1.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-08-14 05:03:41,983 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpd1zsb64i/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpd1zsb64i/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AccessApproval:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpd1zsb64i/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AccessApproval']' returned non-zero exit status 1.
2020-08-14 05:03:42,328 autosynth [ERROR] > Synthesis failed
2020-08-14 05:03:42,328 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 690, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 539, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 630, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/access_approval/synth.metadata', 'synth.py', '--', 'AccessApproval']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/6903d509-9beb-49ce-a5be-156fb55e8d80/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

--- a/config/apis.json
+++ b/config/apis.json
@@ -716,11 +716,6 @@
   },
   {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/plus/v1/rest",
-    "name": "Plus"
-  },
-  {
-    "version": "v1",
     "url": "https://policytroubleshooter.googleapis.com/$discovery/rest?version=v1",
     "name": "PolicyTroubleshooter"
   },

--- a/config/apis.json
+++ b/config/apis.json
@@ -506,6 +506,11 @@
   },
   {
     "version": "v1",
+    "url": "https://www.googleapis.com/discovery/v1/apis/webfonts/v1/rest",
+    "name": "Fonts"
+  },
+  {
+    "version": "v1",
     "url": "https://games.googleapis.com/$discovery/rest?version=v1",
     "name": "Games"
   },
@@ -670,6 +675,11 @@
     "name": "NetworkManagement"
   },
   {
+    "version": "v2",
+    "url": "https://www.googleapis.com/discovery/v1/apis/oauth2/v2/rest",
+    "name": "OAuth2"
+  },
+  {
     "version": "v1",
     "url": "https://osconfig.googleapis.com/$discovery/rest?version=v1",
     "name": "OSConfig"
@@ -723,11 +733,6 @@
     "version": "v1",
     "url": "https://poly.googleapis.com/$discovery/rest?version=v1",
     "name": "Poly"
-  },
-  {
-    "version": "v1alpha1",
-    "url": "https://prod-tt-sasportal.googleapis.com/$discovery/rest?version=v1alpha1",
-    "name": "Prod_TT_SASPortal"
   },
   {
     "version": "v1",
@@ -943,11 +948,6 @@
     "version": "v1",
     "url": "https://tpu.googleapis.com/$discovery/rest?version=v1",
     "name": "TPU"
-  },
-  {
-    "version": "v2",
-    "url": "https://trafficdirector.googleapis.com/$discovery/rest?version=v2",
-    "name": "TrafficDirector"
   },
   {
     "version": "v2",

--- a/config/apis.json
+++ b/config/apis.json
@@ -15,11 +15,6 @@
     "name": "AccessApproval"
   },
   {
-    "version": "v1beta1",
-    "url": "https://accessapproval.googleapis.com/$discovery/rest?version=v1beta1",
-    "name": "AccessApproval"
-  },
-  {
     "version": "v1",
     "url": "https://accesscontextmanager.googleapis.com/$discovery/rest?version=v1",
     "name": "AccessContextManager"
@@ -40,8 +35,8 @@
     "name": "AdExperienceReport"
   },
   {
-    "version": "reports_v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/admin/reports_v1/rest",
+    "version": "datatransfer_v1",
+    "url": "https://www.googleapis.com/discovery/v1/apis/admin/datatransfer_v1/rest",
     "name": "Admin"
   },
   {
@@ -50,8 +45,8 @@
     "name": "Admin"
   },
   {
-    "version": "datatransfer_v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/admin/datatransfer_v1/rest",
+    "version": "reports_v1",
+    "url": "https://www.googleapis.com/discovery/v1/apis/admin/reports_v1/rest",
     "name": "Admin"
   },
   {
@@ -61,12 +56,12 @@
   },
   {
     "version": "v1.4",
-    "url": "https://www.googleapis.com/discovery/v1/apis/adsense/v1.4/rest",
+    "url": "https://adsense.googleapis.com/$discovery/rest?version=v1.4",
     "name": "AdSense"
   },
   {
     "version": "v4.1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/adsensehost/v4.1/rest",
+    "url": "https://adsensehost.googleapis.com/$discovery/rest?version=v4.1",
     "name": "AdSenseHost"
   },
   {
@@ -120,6 +115,11 @@
     "name": "AppsActivity"
   },
   {
+    "version": "v1beta1",
+    "url": "https://artifactregistry.googleapis.com/$discovery/rest?version=v1beta1",
+    "name": "ArtifactRegistry"
+  },
+  {
     "version": "v2",
     "url": "https://www.googleapis.com/discovery/v1/apis/bigquery/v2/rest",
     "name": "BigQuery"
@@ -156,7 +156,7 @@
   },
   {
     "version": "v3",
-    "url": "https://www.googleapis.com/discovery/v1/apis/blogger/v3/rest",
+    "url": "https://blogger.googleapis.com/$discovery/rest?version=v3",
     "name": "Blogger"
   },
   {
@@ -175,8 +175,13 @@
     "name": "Chat"
   },
   {
+    "version": "v1",
+    "url": "https://chromeuxreport.googleapis.com/$discovery/rest?version=v1",
+    "name": "ChromeUXReport"
+  },
+  {
     "version": "v2",
-    "url": "https://www.googleapis.com/discovery/v1/apis/civicinfo/v2/rest",
+    "url": "https://civicinfo.googleapis.com/$discovery/rest?version=v2",
     "name": "CivicInfo"
   },
   {
@@ -285,11 +290,6 @@
     "name": "CloudTrace"
   },
   {
-    "version": "v1alpha1",
-    "url": "https://commentanalyzer.googleapis.com/$discovery/rest?version=v1alpha1",
-    "name": "CommentAnalyzer"
-  },
-  {
     "version": "v1",
     "url": "https://composer.googleapis.com/$discovery/rest?version=v1",
     "name": "Composer"
@@ -326,7 +326,7 @@
   },
   {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/customsearch/v1/rest",
+    "url": "https://customsearch.googleapis.com/$discovery/rest?version=v1",
     "name": "CustomSearch"
   },
   {
@@ -338,6 +338,11 @@
     "version": "v1b3",
     "url": "https://dataflow.googleapis.com/$discovery/rest?version=v1b3",
     "name": "Dataflow"
+  },
+  {
+    "version": "v1",
+    "url": "https://datafusion.googleapis.com/$discovery/rest?version=v1",
+    "name": "DataFusion"
   },
   {
     "version": "v1beta1",
@@ -396,13 +401,18 @@
   },
   {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/dns/v1/rest",
+    "url": "https://dns.googleapis.com/$discovery/rest?version=v1",
     "name": "DNS"
   },
   {
     "version": "v1",
     "url": "https://docs.googleapis.com/$discovery/rest?version=v1",
     "name": "Docs"
+  },
+  {
+    "version": "v1beta2",
+    "url": "https://documentai.googleapis.com/$discovery/rest?version=v1beta2",
+    "name": "DocumentAI"
   },
   {
     "version": "v1",
@@ -460,9 +470,19 @@
     "name": "FirebaseDynamicLinks"
   },
   {
+    "version": "v1",
+    "url": "https://firebasehosting.googleapis.com/$discovery/rest?version=v1",
+    "name": "FirebaseHosting"
+  },
+  {
     "version": "v1beta1",
     "url": "https://firebasehosting.googleapis.com/$discovery/rest?version=v1beta1",
     "name": "FirebaseHosting"
+  },
+  {
+    "version": "v1",
+    "url": "https://firebaseml.googleapis.com/$discovery/rest?version=v1",
+    "name": "FirebaseML"
   },
   {
     "version": "v1",
@@ -481,23 +501,23 @@
   },
   {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/fitness/v1/rest",
+    "url": "https://www.googleapis.com/$discovery/rest?version=v1",
     "name": "Fitness"
   },
   {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/webfonts/v1/rest",
-    "name": "Fonts"
-  },
-  {
-    "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/games/v1/rest",
+    "url": "https://games.googleapis.com/$discovery/rest?version=v1",
     "name": "Games"
   },
   {
     "version": "v1configuration",
-    "url": "https://www.googleapis.com/discovery/v1/apis/gamesConfiguration/v1configuration/rest",
+    "url": "https://gamesconfiguration.googleapis.com/$discovery/rest?version=v1configuration",
     "name": "GamesConfiguration"
+  },
+  {
+    "version": "v1",
+    "url": "https://gameservices.googleapis.com/$discovery/rest?version=v1",
+    "name": "GameServices"
   },
   {
     "version": "v1beta",
@@ -506,7 +526,7 @@
   },
   {
     "version": "v1management",
-    "url": "https://www.googleapis.com/discovery/v1/apis/gamesManagement/v1management/rest",
+    "url": "https://gamesmanagement.googleapis.com/$discovery/rest?version=v1management",
     "name": "GamesManagement"
   },
   {
@@ -520,14 +540,24 @@
     "name": "Gmail"
   },
   {
+    "version": "v1beta1",
+    "url": "https://gmailpostmastertools.googleapis.com/$discovery/rest?version=v1beta1",
+    "name": "GmailPostmasterTools"
+  },
+  {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/groupsmigration/v1/rest",
+    "url": "https://www.googleapis.com/$discovery/rest?version=v1",
     "name": "GroupsMigration"
   },
   {
     "version": "v1",
     "url": "https://www.googleapis.com/discovery/v1/apis/groupssettings/v1/rest",
     "name": "GroupsSettings"
+  },
+  {
+    "version": "v1",
+    "url": "https://healthcare.googleapis.com/$discovery/rest?version=v1",
+    "name": "HealthCare"
   },
   {
     "version": "v1beta1",
@@ -615,6 +645,11 @@
     "name": "Manufacturers"
   },
   {
+    "version": "v1",
+    "url": "https://memcache.googleapis.com/$discovery/rest?version=v1",
+    "name": "Memcache"
+  },
+  {
     "version": "v1beta2",
     "url": "https://memcache.googleapis.com/$discovery/rest?version=v1beta2",
     "name": "Memcache"
@@ -625,14 +660,14 @@
     "name": "Monitoring"
   },
   {
-    "version": "v1beta1",
-    "url": "https://networkmanagement.googleapis.com/$discovery/rest?version=v1beta1",
+    "version": "v1",
+    "url": "https://networkmanagement.googleapis.com/$discovery/rest?version=v1",
     "name": "NetworkManagement"
   },
   {
-    "version": "v2",
-    "url": "https://www.googleapis.com/discovery/v1/apis/oauth2/v2/rest",
-    "name": "OAuth2"
+    "version": "v1beta1",
+    "url": "https://networkmanagement.googleapis.com/$discovery/rest?version=v1beta1",
+    "name": "NetworkManagement"
   },
   {
     "version": "v1",
@@ -651,7 +686,7 @@
   },
   {
     "version": "v5",
-    "url": "https://www.googleapis.com/discovery/v1/apis/pagespeedonline/v5/rest",
+    "url": "https://pagespeedonline.googleapis.com/$discovery/rest?version=v5",
     "name": "PageSpeedOnline"
   },
   {
@@ -660,9 +695,19 @@
     "name": "People"
   },
   {
+    "version": "v3",
+    "url": "https://playablelocations.googleapis.com/$discovery/rest?version=v3",
+    "name": "PlayableLocations"
+  },
+  {
     "version": "v1",
-    "url": "https://www.googleapis.com/discovery/v1/apis/playcustomapp/v1/rest",
+    "url": "https://www.googleapis.com/$discovery/rest?version=v1",
     "name": "PlayCustomApp"
+  },
+  {
+    "version": "v1",
+    "url": "https://www.googleapis.com/discovery/v1/apis/plus/v1/rest",
+    "name": "Plus"
   },
   {
     "version": "v1",
@@ -680,9 +725,34 @@
     "name": "Poly"
   },
   {
+    "version": "v1alpha1",
+    "url": "https://prod-tt-sasportal.googleapis.com/$discovery/rest?version=v1alpha1",
+    "name": "Prod_TT_SASPortal"
+  },
+  {
     "version": "v1",
     "url": "https://pubsub.googleapis.com/$discovery/rest?version=v1",
     "name": "PubSub"
+  },
+  {
+    "version": "v1",
+    "url": "https://pubsublite.googleapis.com/$discovery/rest?version=v1",
+    "name": "PubSubLite"
+  },
+  {
+    "version": "v1",
+    "url": "https://realtimebidding.googleapis.com/$discovery/rest?version=v1",
+    "name": "RealTimeBidding"
+  },
+  {
+    "version": "v1beta1",
+    "url": "https://recommendationengine.googleapis.com/$discovery/rest?version=v1beta1",
+    "name": "RecommendationEngine"
+  },
+  {
+    "version": "v1",
+    "url": "https://recommender.googleapis.com/$discovery/rest?version=v1",
+    "name": "Recommender"
   },
   {
     "version": "v1beta1",
@@ -725,6 +795,11 @@
     "name": "SafeBrowsing"
   },
   {
+    "version": "v1alpha1",
+    "url": "https://sasportal.googleapis.com/$discovery/rest?version=v1alpha1",
+    "name": "SASPortal"
+  },
+  {
     "version": "v1",
     "url": "https://script.googleapis.com/$discovery/rest?version=v1",
     "name": "Script"
@@ -757,6 +832,11 @@
   {
     "version": "v1",
     "url": "https://servicecontrol.googleapis.com/$discovery/rest?version=v1",
+    "name": "ServiceControl"
+  },
+  {
+    "version": "v2",
+    "url": "https://servicecontrol.googleapis.com/$discovery/rest?version=v2",
     "name": "ServiceControl"
   },
   {
@@ -855,19 +935,19 @@
     "name": "TextToSpeech"
   },
   {
-    "version": "v1",
-    "url": "https://toolresults.googleapis.com/$discovery/rest?version=v1",
-    "name": "ToolResults"
-  },
-  {
     "version": "v1beta3",
-    "url": "https://www.googleapis.com/discovery/v1/apis/toolresults/v1beta3/rest",
+    "url": "https://toolresults.googleapis.com/$discovery/rest?version=v1beta3",
     "name": "ToolResults"
   },
   {
     "version": "v1",
     "url": "https://tpu.googleapis.com/$discovery/rest?version=v1",
     "name": "TPU"
+  },
+  {
+    "version": "v2",
+    "url": "https://trafficdirector.googleapis.com/$discovery/rest?version=v2",
+    "name": "TrafficDirector"
   },
   {
     "version": "v2",
@@ -883,6 +963,11 @@
     "version": "v1",
     "url": "https://vault.googleapis.com/$discovery/rest?version=v1",
     "name": "Vault"
+  },
+  {
+    "version": "v1",
+    "url": "https://vectortile.googleapis.com/$discovery/rest?version=v1",
+    "name": "VectorTile"
   },
   {
     "version": "v1",

--- a/config/exceptions.json
+++ b/config/exceptions.json
@@ -1,0 +1,22 @@
+[
+  {
+    "type": "omit",
+    "name": "trafficdirector",
+    "version": "v2"
+  },
+  {
+    "type": "omit",
+    "name": "prod_tt_sasportal",
+    "version": "v1alpha1"
+  },
+  {
+    "type": "keep",
+    "name": "OAuth2",
+    "version": "v2"
+  },
+  {
+    "type": "keep",
+    "name": "Fonts",
+    "version": "v1"
+  }
+]

--- a/config/exceptions.json
+++ b/config/exceptions.json
@@ -1,6 +1,11 @@
 [
   {
     "type": "omit",
+    "name": "plus",
+    "version": "v1"
+  },
+  {
+    "type": "omit",
     "name": "trafficdirector",
     "version": "v2"
   },

--- a/lib/google_apis/discovery.ex
+++ b/lib/google_apis/discovery.ex
@@ -40,9 +40,10 @@ defmodule GoogleApis.Discovery do
   end
 
   def discover_merge() do
+    exceptions = get_exceptions()
     ApiConfig.load_all()
-    |> remove_entries(discover(preferred: false))
-    |> add_entries(discover(preferred: true))
+    |> remove_entries(discover(preferred: false, exceptions: exceptions), exceptions: exceptions)
+    |> add_entries(discover(preferred: true, exceptions: exceptions))
     |> Enum.sort_by(fn config ->
       {config.publish == false, String.downcase(config.name), config.version, config.url}
     end)
@@ -55,18 +56,36 @@ defmodule GoogleApis.Discovery do
   def discover(opts \\ []) do
     conn = Connection.new()
     preferred = Keyword.get(opts, :preferred, true)
+    exceptions = Keyword.get_lazy(opts, :exceptions, &get_exceptions/0)
     {:ok, %{items: items}} = Apis.discovery_apis_list(conn, preferred: preferred)
 
-    Enum.map(items, fn item ->
-      %ApiConfig{
-        name: item.name,
-        version: item.version,
-        url: item.discoveryRestUrl
-      }
+    Enum.flat_map(items, fn %{name: name, version: version, discoveryRestUrl: url} ->
+      if Enum.any?(exceptions, fn
+        %{type: "omit", name: ^name, version: ^version} -> true
+        _ -> false
+      end) do
+        []
+      else
+        [%ApiConfig{name: name, version: version, url: url}]
+      end
     end)
   end
 
-  defp remove_entries(current, discovered) do
+  defmodule Exception do
+    defstruct type: nil,
+              name: nil,
+              version: nil
+  end
+
+  defp get_exceptions() do
+    "./config/exceptions.json"
+    |> Path.expand()
+    |> File.read!()
+    |> Poison.decode!(as: [%Exception{}])
+  end
+
+  defp remove_entries(current, discovered, opts) do
+    exceptions = Keyword.get_lazy(opts, :exceptions, &get_exceptions/0)
     Enum.map(current, fn
       (%ApiConfig{publish: false} = cur_config) ->
         cur_config
@@ -86,7 +105,14 @@ defmodule GoogleApis.Discovery do
         end)
         |> case do
           nil ->
-            %ApiConfig{cur_config | publish: "CHECKME: No longer in discovery. Consider removal."}
+            if Enum.any?(exceptions, fn
+              %{type: "keep", name: ^cur_name, version: ^cur_version} -> true
+              _ -> false
+            end) do
+              cur_config
+            else
+              %ApiConfig{cur_config | publish: "CHECKME: No longer in discovery. Consider removal."}
+            end
           altered_config ->
             altered_config
         end

--- a/lib/google_apis/discovery.ex
+++ b/lib/google_apis/discovery.ex
@@ -67,25 +67,61 @@ defmodule GoogleApis.Discovery do
   end
 
   defp remove_entries(current, discovered) do
-    Enum.map(current, fn %ApiConfig{url: url, publish: publish} = config ->
-      if publish == false || Enum.any?(discovered, fn c -> c.url == url end) do
-        config
-      else
-        %ApiConfig{config | publish: "FIXME: No longer in discovery. Consider removal."}
-      end
+    Enum.map(current, fn
+      (%ApiConfig{publish: false} = cur_config) ->
+        cur_config
+      (%ApiConfig{url: cur_url, version: cur_version, name: cur_name} = cur_config) ->
+        discovered
+        |> Enum.find_value(fn
+          %ApiConfig{url: ^cur_url} ->
+            cur_config
+          %ApiConfig{version: ^cur_version, name: new_name, url: new_url} ->
+            if String.downcase(new_name) == String.downcase(cur_name) do
+              %ApiConfig{cur_config | url: new_url, publish: "CHECKME: Url changed from #{cur_url}"}
+            else
+              false
+            end
+          _ ->
+            false
+        end)
+        |> case do
+          nil ->
+            %ApiConfig{cur_config | publish: "CHECKME: No longer in discovery. Consider removal."}
+          altered_config ->
+            altered_config
+        end
     end)
   end
 
   defp add_entries(current, discovered) do
-    added = Enum.flat_map(discovered, fn config ->
-      url = config.url
-      if Enum.any?(current, fn c -> c.url == url end) do
-        []
-      else
-        [%ApiConfig{config | publish: "FIXME: Newly added. Adjust the name capitalization."}]
+    Enum.reduce(discovered, current, fn(%ApiConfig{name: new_name, url: new_url, version: new_version} = new_config, configs) ->
+      configs
+      |> Enum.map_reduce(new_name, fn
+        (cur_config, nil) ->
+          {cur_config, nil}
+        (%ApiConfig{url: ^new_url} = cur_config, _) ->
+          {cur_config, nil}
+        (%ApiConfig{url: cur_url, name: cur_name, version: cur_version} = cur_config, working_name) ->
+          if String.downcase(cur_name) == String.downcase(working_name) do
+            if cur_version == new_version do
+              changed_config = %ApiConfig{cur_config | url: new_url, publish: "CHECKME: Url changed from #{cur_url}"}
+              {changed_config, nil}
+            else
+              {cur_config, cur_name}
+            end
+          else
+            {cur_config, working_name}
+          end
+      end)
+      |> case do
+        {modified_configs, nil} ->
+          modified_configs
+        {modified_configs, new_name} ->
+          new_name = if String.downcase(new_name) == new_name, do: String.capitalize(new_name), else: new_name
+          nc = %ApiConfig{new_config | name: new_name, publish: "CHECKME: Newly added. Adjust the name capitalization."}
+          [nc | modified_configs]
       end
     end)
-    current ++ added
   end
 
   defp format_entries(entries) do

